assets/app/view/game/buy_trains.rb

```diff
@@ -41,7 +41,10 @@ def render_president_contributions
             children << h(:div, props, corp)
           end
 
-          children << render_bankruptcy
+          round = @game.round
+          corporation = round.current_entity
+          current_actions = round.actions_for(corporation)
+          children << render_bankruptcy unless current_actions.include?('buy_shares')
```

Comment:
why if you can't buy shares?

Reply:
In 1846 the president can't sell shares for EMR before the corporation has issued shares. The IssueShares view is rendered if ` buy_shares` is an available action:

In https://github.com/tobymao/18xx/blob/master/assets/app/view/game/round/operating.rb#L28

I presume `buy_shares` was used there since `sell_shares` could happen during two different steps--IssueShares and Train.

I wanted to avoid declaring bankruptcy until after shares were issued, so this doesn't render the Declare Bankruptcy button if `buy_shares` is available--really I want to not render that button if the _corporation_ can sell shares, but that's handled by the same available action as the _president_ selling shares. Another approach would be to make `process_bankrupt` do that, similar to how it handles the other share selling the president does.

Reply:
f8defb66
6a3344ad